skbio/sequence/_sequence.py

```diff
@@ -522,7 +528,7 @@ def _positional_metadata_axis_len_(self):
 
     @stable(as_of="0.4.0")
     def __init__(self, sequence, metadata=None, positional_metadata=None,
-                 lowercase=False):
+                 lowercase=False, features=None):
```

Comment:
rename features to `interval_metadata` and place it next to `positional_metadata`?